test/Base_Tests/src/Semantic/Meta_Spec.enso

```diff
@@ -89,6 +90,24 @@ add_specs suite_builder =
         group_builder.specify "should allow getting a value's constructor's fields" <|
             Meta.meta List.Nil . constructor . fields . should_equal []
             Meta.meta (List.Cons 1 List.Nil) . constructor . fields . should_equal ["x", "xs"]
+
+        group_builder.specify "prevent getting value from private constructor" <|
+            atom = Public_Type.new 6*7
+            meta = Meta.meta atom
+
+            # fields are invisible
+            meta.fields . should_equal []
+
+        group_builder.specify "prevent getting fields from private constructor" <|
+            atom = Public_Type.new 6*7
+            meta = Meta.meta atom
+
+            # constructor is inaccessible
+            meta.constructor . should_fail_with Private_Access
+
+            # fields should be invisible
+            meta.constructor.fields . should_fail_with Private_Access
```

Comment:
The comment and behaviour tested seem inconsistent.

Reply:
It is not completely inconsistent. The value of `meta.constructor` is `Error` thus calling `.fields` on it is also automatically an error and fields remain invisible. Anyway, comment changed in 847e6ef85e